pkg/controller/cloud/node_controller.go

```diff
@@ -125,6 +155,63 @@ func (cnc *CloudNodeController) UpdateNodeStatus(ctx context.Context) {
 	for i := range nodes.Items {
 		cnc.updateNodeAddress(ctx, &nodes.Items[i], instances)
 	}
+
+	for _, node := range nodes.Items {
+		err = cnc.reconcileNodeLabels(node.Name)
```

Comment:
Would it make sense to re-use the loop from line 155 and avoid "getting" the node object again in line 169?